File: BOJ_1085/BOJ_1085.cs
using System;
namespace escape_from_rectangle{
        class Program{
                static void Main(string[] args){
                        var line = Console.ReadLine();
                        var data = line.Split(' ');
                        int X = int.Parse(data[0]);
                        int Y = int.Parse(data[1]);
                        int W = int.Parse(data[2]);
                        int H = int.Parse(data[3]);
                        int TO_BORDER_X = W-X;
                        int TO_BORDER_Y = H-Y;
                        int PLACE_HOLDER = Math.Min(X,Y);
                        PLACE_HOLDER = Math.Min(PLACE_HOLDER, TO_BORDER_X);
                        PLACE_HOLDER = Math.Min(PLACE_HOLDER, TO_BORDER_Y);
                        Console.WriteLine(PLACE_HOLDER);

                }
        }
}

File: BOJ_1085/BOJ_1085.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
class Main{
        public static void main(String[] args)
        throws IOException{
        BufferedReader BR = new BufferedReader(new InputStreamReader(System.in));
        String[] strNums;
        strNums = BR.readLine().split(" ");
        int X = Integer.parseInt(strNums[0]);
        int Y = Integer.parseInt(strNums[1]);
        int W = Integer.parseInt(strNums[2]);
        int H = Integer.parseInt(strNums[3]);
        int GAP_X = W-X;
        int GAP_Y = H-Y;
        int PLACE_HOLDER = Math.min(Math.min(X, Y), Math.min(GAP_X, GAP_Y));
        System.out.println(PLACE_HOLDER);
        }
}
